tests/services/google_sheets_service.py

```
@@ -58,9 +59,15 @@ def get_sheet_headers(spreadsheet_name, sheet_name):
-    client = get_google_sheets_client()
-    sheet = client.open(spreadsheet_name).worksheet(sheet_name)
-    return sheet.get_all_records(expected_headers=expected_headers)
+    try:
+        client = get_google_sheets_client()
+        sheet = client.open(spreadsheet_name).worksheet(sheet_name)
+        return sheet.get_all_records(expected_headers=expected_headers)
+    except SpreadsheetNotFound as e:
+        logger.error(e)
+        raise ValueError(
+            f"Spreadsheet '{spreadsheet_name}' not found. Check if it's not XLSX or try publish it for service account."
+        )
```

Ruff

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)